src/GatewayAPI/Database/LedgerStateQuerier.cs

```diff
@@ -139,10 +139,10 @@ public async Task<GatewayResponse> GetGatewayState()
     }
 
     // So that we don't forget to check the network name, add the assertion in here.
-    public async Task<LedgerState> GetValidLedgerStateForReadRequest(NetworkIdentifier networkIdentifier, PartialLedgerStateIdentifier? atLedgerStateIdentifier)
+    public async Task<LedgerState> GetValidLedgerStateForReadRequest(NetworkIdentifier networkIdentifier, PartialLedgerStateIdentifier? atLedgerStateIdentifier, bool ascending = false)
```

Comment:
I fear we're slightly muddling two ideas here, which may cause a confusing interface. (EG right now, leaving out `atLedgerStateIdentifier` for ascending requests gives an empty response I think?)

I think, for consistency, there's actually two parameters we need here:

* `PartialLedgerStateIdentifier? atLedgerStateIdentifier`
* `PartialLedgerStateIdentifier? fromLedgerStateIdentifier`

The `atLedgerStateIdentifier` should be consistent with the rest of the API. Therefore, it determines the historic context of the request - basically the max state version available. This will normally not be provided as a request parameter, but we'll still resolve it in the ledgerStateReport, and use it as context in all API requests made. This essentially provides the cap you were hinting at on the ticket, to allow us to return a null cursor for ascending transaction streams.

The `fromLedgerStateIdentifier` would provide the ability for the user to say "what state version should my request start from" (asc or desc). This is the one where we'd pass in the `GetNextLedgerState(fromLedgerStateIdentifier, ascending/descending, atLedgerStateIdentifier)`. If not provided, ASC should default to state version 1 (ie from genesis), and DESC should default to the ledgerStateReport.